src/Patient_Dashboard/UploadDocumentPage.js

```diff
@@ -0,0 +1,289 @@
+            );
+          }
+        } catch (error) {
+          console.error("Error uploading document:", error);
+          reject(error);
+        }
+      };
+    });
+  } catch (error) {
+    console.error("Error reading file:", error);
+    throw error;
+  }
+};
+
+const UploadDocumentPage = () => {
+  const { visitId } = useParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { patientData, visit, selectedDocumentPreview, visitDate } =
+    location.state  || {};
+  const uploadedVisitId = visit ? visit.uuid : null;
+
+  const formatVisitDate = (datetime) => {
+    if (isValid(new Date(datetime))) {
+      return format(new Date(datetime), "MMM dd, yyyy hh:mm a");
+    } else {
+      return "Invalid Date";
+    }
+  };
+  const patientName = patientData ? patientData.name : "";
+
+  const [selectedDocument, setSelectedDocument] = useState(null);
+  const [documentPreviews, setDocumentPreviews] = useState([]);
+  const [selectedFileType, setSelectedFileType] = useState(null);
+
+  const handleDocumentUploadIcon = () => {
+    const fileInput = document.getElementById(`fileInput`);
+    if (fileInput) {
+      fileInput.click();
+    }
+  };
+
+  const handleDocumentFileChange = (event) => {
+    const file = event.target.files[0];
+    if (file) {
+      setSelectedDocument(file);
+      const previewUrl = URL.createObjectURL(file); // Create a preview URL
+      setDocumentPreviews((prevPreviews) => [...prevPreviews, previewUrl]); // Add to previews
+    }
+  };
+
+  const handleCapture = () => {
+    console.log("Capture clicked");
+    const fileInput = document.createElement("input");
+    fileInput.type = "file";
+    fileInput.accept = "application/pdf,image/*";
+    fileInput.capture = "environment"; // Use the back camera for mobile devices
+    fileInput.onchange = (event) => handleDocumentFileChange(event);
+    fileInput.click();
+  };
+
+  const handleCancel = () => {
+    navigate(`/patient_dashboard/${patientData.uuid}`, {
+      state: {
+        patientData,
+      },
+    });
+  };
+
+  const handleSaveAndUpload = async () => {
+    if (!selectedDocument) {
+      alert("Please select a document to upload.");
+      return;
+    }
+
+    try {
+      const response = await saveFile(
+        selectedDocument,
+        patientData.uuid,
+        selectedFileType, // Use the selected file type
+        selectedDocument.name
+      );
+
+      console.log("Document uploaded successfully");
+      console.log("File URL:", response);
+      alert("Document(s) saved and uploaded successfully");
+      navigate(`/patient_dashboard/${patientData.uuid}`, {
+        state: {
+          patientData,
+        },
+      });    } catch (error) {
+      console.error("Error uploading document:", error);
+    }
+  };
+
+  useEffect(() => {
+    if (selectedDocumentPreview) {
+      setDocumentPreviews([selectedDocumentPreview]);
+    }
+  }, []);
+
+  return (
+    <DashboardLayout>
```

Comment:
A huge snippet of HTML. Would prefer composing multiple react components to create the final subpage, instead of a long snippet of HTML code.